### ISSUE TITLE
Extend ESLint rule object-curly-spacing for TS interfaces

### ISSUE THREAD

**Issue by essenmitsosse** [typescript-eslint/typescript-eslint#2215]
**Repro**
```TS
interface Foo {a: number}
interface Bar { a: number }
```

**Expected Result**
Should throw an error, to enforces consistent spacing inside braces of interfaces, like [object-curly-spacing](https://eslint.org/docs/rules/object-curly-spacing) does for objects. Auto fixing would be great as well.

**Actual Result**
Throws no error — rules doesn't exist yet.

**Additional Info**
Would be willing to give a shot at a PR, if that feature would be welcome.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.2.0` |
| `@typescript-eslint/parser`        | `3.2.0` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `13.14.0` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
Happy to accept a PR!

**Comment by essenmitsosse** [comment]
Ok good to know, I have a look at prettier again first. I just found this seems to be the only styling rule that is missing in TypeScript declarations, thats works in ESLint for non-TypeScript stuff.

**Comment by essenmitsosse** [comment]
Gave prettier a try and definitely agree with you that you should let prettier handle the code formatting. Integration with eslint is fantastic.
